content/content/oop_macro.md

```diff
@@ -154,33 +154,37 @@ macro class*(head: expr, body: stmt): stmt {.immediate.} =
   #             Empty
 
   result.insert(0,
-    if baseName == nil:
-      quote do:
-        type `typeName` = ref object of RootObj
-    else:
-      quote do:
-        type `typeName` = ref object of `baseName`
+    newNimNode(nnkTypeSection).add(
```

Comment:
i can do this:
```
  # create an identifier for the base object
  if baseName == nil:
    baseName = newIdentNode("RootObj")

  result.insert(0,
    quote do:
      type `typeName` = ref object of `baseName`
  )
```
Or I could move the if in the first branch at the top, like so:
```
  if head.kind == nnkIdent:
    # `head` is expression `typeName`
    # echo head.treeRepr
    # --------------------
    # Ident !"Animal"
    typeName = head
    baseName = newIdentNode("RootObj")

```
which one would you prefer?